File: app/Python/src/br/com/monkeyconsulting/infra/database/models/cliente_model.py
from sqlalchemy import Column, String, Integer, ForeignKey, Boolean
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import relationship

from br.com.monkeyconsulting.domain.dtos.cliente_dto import ClienteDTO
from br.com.monkeyconsulting.infra.database.models.datas_model import DataModel
from src.br.com.monkeyconsulting.infra.database.models.dietas_treinos_model import DietaTreinoModel
from src.br.com.monkeyconsulting.infra.database.models.planos_model import PlanoModel

Base = declarative_base()


class ClienteModel(Base):
    __tablename__ = 'CLIENTES'

    nome = Column(String, nullable=False)
    telefone = Column(String, nullable=False)
    indicador_cliente_ativo = Column(Boolean, nullable=False)

    id_cliente = Column(Integer, primary_key=True)
    id_plano = Column(Integer, ForeignKey(PlanoModel.id_plano), nullable=False)
    id_dieta = Column(Integer, ForeignKey(DietaTreinoModel.id_dieta), nullable=False)
    id_data = Column(Integer, ForeignKey(DataModel.id_data), nullable=False)

    plano = relationship(PlanoModel, lazy=False)
    dieta = relationship(DietaTreinoModel)
    data = relationship(DataModel)

    def to_model(self, dto: ClienteDTO) -> 'ClienteModel':
        self.id_cliente = dto.id
        self.nome = dto.nome
        self.telefone = dto.telefone
        self.indicador_cliente_ativo = dto.indicador_cliente_ativo
        self.id_plano = dto.plano.id_plano if dto.plano is not None else None
        self.id_dieta = dto.dieta.id_dieta if dto.dieta is not None else None
        self.id_data = dto.data.id_data if dto.data is not None else None
        return self

    def to_dto(self):
        dto = ClienteDTO()
        dto.id = self.id_cliente
        dto.nome = self.nome
        dto.telefone = self.telefone
        dto.indicador_cliente_ativo = self.indicador_cliente_ativo
        dto.plano = self.plano.to_dto() if self.plano is not None else None
        dto.dieta = self.dieta.to_dto() if self.dieta is not None else None
        dto.data = self.data.to_dto() if self.data is not None else None
        return dto

    def __repr__(self):
        return f"<ClienteModel(id='{self.id_cliente}')>"

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

File: app/src/main/java/br/com/monkeyconsulting/models/ClienteModel.java
package br.com.monkeyconsulting.models;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;


//@EqualsAndHashCode(callSuper = true)
@Entity(name = "CLIENTES")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id_cliente")
@Data
public class ClienteModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cliente")
    @JsonProperty("id_cliente")
    private Integer idCliente;

    @Column(name = "nome")
    @JsonProperty("nome")
    private String nome;

    @Column(name = "telefone")
    @JsonProperty("telefone")
    private Long telefone;

    @Column(name = "indicador_cliente_ativo")
    @JsonProperty("indicador_cliente_ativo")
    private Boolean indicadorClienteAtivo;

    @ManyToOne
    @JsonProperty("plano")
    @JoinColumn(name = "id_plano")
    private PlanoModel planoModel;

    @ManyToOne
    @JsonProperty("dieta")
    @JoinColumn(name = "id_dieta")
    private DietaTreinoModel dietaTreinoModel;

    @JsonProperty("valores")
    @OneToMany(mappedBy = "clientesModel")
    @JsonManagedReference
    private List<ValorModel> valoresModel;

    @JsonProperty("datas")
    @OneToMany(mappedBy = "clientesModel")
    @JsonManagedReference
    private List<DataModel> datasModel;

    public void mergeModels(DataModel datas, ValorModel valor){
        this.datasModel.add(datas);
        this.valoresModel.add(valor);
    }

}
